Network integration test

justinta opened the pull request:
### What does this PR do?

Adds tests for the network state module. Checks that typical behavior works as expected. 
### What issues does this PR fix or reference?

N/A
### Tests written?

Yes

Commit:
return list of nodes for lxc driver when called directly

This is required so that salt-cloud sees a dictionary list of lxc
containers that exist so that they do not attempt to recreate the
container.

The downside to this is that if the container is stopped, it is not
started.

Fixes #33633

Commit:
Integration test for network state

Commit:
Integration test for network state

Commit:
Removed unused imports.

Commit:
Removed unused imports.

ghedo committed:
Do not force 'filter' table when flushing

The "table" argument is already part of the function signature, this means
that flush() will always force the "filter" table even when the user sets
a different one.

Commit:
Fix test example in integration testing docs

The example is out of date compared to the actual test it's modeling.

Notably, the self.state_result() function doesn't exist.

Commit:
Changed network state to use sls files. Added sls test files.

Commit:
Changed network state to use sls files. Added sls test files.

Commit:
set function return to ret in test_system

Commit:
set function return to ret in test_system

Commit:
Merge pull request #34307 from rallytime/fix-test-example

Fix test example in integration testing docs

cachedout commented on tests/integration/states/network.py:
```diff
@@ -0,0 +1,81 @@
+
+
+@destructiveTest
+@skipIf(_check_arch_linux() == 'Arch Linux', 'Network state not support on Arch')
```
`not supported`

Commit:
[2015.5] Update to latest bootstrap script v2016.06.27

Commit:
[2015.8] Update to latest bootstrap script v2016.06.27

cachedout commented on tests/integration/states/network.py:
```diff
@@ -0,0 +1,81 @@
+        '''
+        ret = self.run_function('state.sls', mods='network.system')
+
+        self.assertSaltTrueReturn(ret)
```
While these sorts of tests do exercise the execution modules that are called by the state, they don't do much in terms of verifying that we've accomplished the given task. Could we at least examine the `changes` dictionary to ensure it contains the expected data?

Commit:
Support to Supported

Commit:
Support to Supported

Commit:
Merge pull request #34312 from rallytime/bootstrap-2015.8

[2015.8] Update to latest bootstrap script v2016.06.27

Commit:
Merge pull request #34313 from rallytime/bootstrap-2015.5

[2015.5] Update to latest bootstrap script v2016.06.27

Commit:
salt.utils.open to salt.utils.fopen

Commit:
salt.utils.open to salt.utils.fopen

Commit:
Typo in dockerio doc (#34319)

Commit:
Merge branch '2015.5' into '2015.8'

No conflicts.

Commit:
Merge pull request #34335 from rallytime/merge-2015.8

[2015.8] Merge forward from 2015.5 to 2015.8

Commit:
Remove unnecessarily-disabled sanity check (#34325)

This was commented out for reasons having nothing to do with anything
else going on in 0c069dd. Uncommenting this sanity check so that we are
notified if we need to update our targets in the future.

jacobhammons committed:
Doc clarifications to file modules, addition of new `profile` log level to docs, fixed example in dnsmasq (#34323)

Refs #34249
Refs #34261
Refs #33694
Refs #34247

Commit:
Merge pull request #34252 from gtmanfred/2015.5

return list of nodes for lxc driver when called directly

Commit:
Merge pull request #34306 from ghedo/iptables_flush_table

Fix iptables.flush state: Do not force 'filter' table when flushing

Commit:
Revert py3modernize lint changes (#34339)

* salt/__init__.py: revert py3modernize lint changes

* salt/cloud/clouds/vmware.py: revert py3modernize lint changes

* salt/modules/jboss7_cli.py: revert py3modernize lint changes

* salt/payload.py: revert py3modernize lint changes

* salt/serializers/yamlex.py: revert py3modernize lint changes

* salt/states/win_servermanager.py: use absolute imports

* salt/utils/args.py: revert py3modernize lint changes

* salt/utils/decorators/__init__.py: use __name__ instead of func_name

__name__ is py3-compatible and also works on builtins in py2, which
func_name does not.

* salt/utils/compat.py: revert py3modernize lint changes

* salt/utils/__init__.py: revert py3modernize lint changes

* salt/version.py: revert py3modernize lint changes

* tests/salt-tcpdump.py: revert py3modernize lint changes

Commit:
Making salt-ssh pass proper return codes for jinja rendering errors (#34342)

Commit:
Back-port #34324 to 2015.8 (#34344)

* Add test to match against custom grains

* Lint

* Forgot to add custom grain

Commit:
Check against changes dict test_routes

Commit:
Check against changes dict test_routes

Commit:
Merge branch '2015.5' into '2015.8'

Conflicts:
  - salt/cloud/clouds/lxc.py

Commit:
Merge pull request #34368 from rallytime/merge-2015.8

[2015.8] Merge forward from 2015.5 to 2015.8

Commit:
Check against changes dict test_system

Commit:
Check against changes dict test_system

Commit:
Merge branch 'network_integration_test' of github.com:jtand/salt into network_integration_test

jtand commented:
Screwed up the rebase. re submitting.
